app/models/compute_resources/foreman/model/gce.rb

```diff
@@ -246,11 +246,18 @@ def construct_network_interfaces(network_interfaces_list, external_ip = nil)
       network_interfaces_list
     end
 
-    def check_google_key_path
+    def check_google_key_format_and_options
       return if key_path.blank?
-      unless File.exist?(key_path)
+      if File.exist?(key_path)
```

Comment:
It may be easier to split the code in two:
```ruby
def read_key_file
  return nil unless File.exist?(key_path)
  JSON.parse(File.read(key_path).chomp
end

def check_google_key_format_and_options
  return if key_path.blank?
  begin
    key_data = read_key_file
    unless key_data.is_a?(Hash) && key_data.key?('client_email') && key_data.key?('private_key')
      # ...
    end
  rescue JSON::ParserError
    # ..
  end
end
```

Now you can just stub your helper method in your tests to raise an error or return some weird value. You can also test that part in particular.